File: src/main/java/com/dh/ondot/member/core/TokenExtractor.java
package com.dh.ondot.member.core;

import com.dh.ondot.member.core.exception.InvalidTokenHeaderException;
import com.dh.ondot.member.core.exception.TokenMissingException;
import lombok.experimental.UtilityClass;

@UtilityClass
public class TokenExtractor {
    private static final String BEARER_PREFIX = "Bearer ";

    public static String extract(String header) {
        if (header == null) {
            throw new TokenMissingException();
        }
        if (!header.startsWith(BEARER_PREFIX)) {
            throw new InvalidTokenHeaderException();
        }
        return header.substring(BEARER_PREFIX.length());
    }
}

File: src/main/kotlin/com/dh/ondot/member/core/TokenExtractor.kt
package com.dh.ondot.member.core

import com.dh.ondot.member.core.exception.InvalidTokenHeaderException
import com.dh.ondot.member.core.exception.TokenMissingException

object TokenExtractor {
    private const val BEARER_PREFIX = "Bearer "

    @JvmStatic
    fun extract(header: String?): String {
        if (header == null) {
            throw TokenMissingException()
        }
        if (!header.startsWith(BEARER_PREFIX)) {
            throw InvalidTokenHeaderException()
        }
        return header.substring(BEARER_PREFIX.length)
    }
}
